controller/appcontroller.go

```diff
@@ -1512,10 +1513,15 @@ func (ctrl *ApplicationController) processAppRefreshQueueItem() (processNext boo
 	}
 	now := metav1.Now()
 
-	compareResult := ctrl.appStateManager.CompareAppState(app, project, revisions, sources,
+	compareResult, err := ctrl.appStateManager.CompareAppState(app, project, revisions, sources,
 		refreshType == appv1.RefreshTypeHard,
 		comparisonLevel == CompareWithLatestForceResolve, localManifests, hasMultipleSources)
 
+	if goerrors.Is(err, CompareStateRepoError) {
+		logCtx.Warnf("Ignoring temporary failed attempt to compare app state against repo: %v", err)
+		return // short circuit if git error is encountered
+	}
+
```

Comment:
Something about introducing an error return value from CompareAppState method signature for the **_sole purpose_** of CompareStateRepoError feels unclean. I think it's pretty specialized, and the fact that we only return error in this circumstance is going to get lost over time.

Instead of introducing error for this one purpose, what if we return `nil`, (which is what you are already doing the PR). The caller should interpret a nil return value as indication comparison should be skipped/short-circuted. This way, it doesn't affect the method signature and the semantics seem cleaner.

Reply:
I initially planned to do that, where we returned nil but then decided against it and made it more specific because in the future if the code changes and there's other cases where nil is returned we might forget to update this logic so I chose to go with a specific error for this case.